guide/yaml/yaml-reference.md

```diff
@@ -213,6 +213,9 @@ concise DSL defined here:
   `$brooklyn:root()`.
 * `$brooklyn:formatString("pattern e.g. %s %s", "field 1", "field 2")` returns a future which creates the formatted string
   with the given parameters, where parameters may be strings *or* other tasks such as `attributeWhenReady`
+* `$brooklyn:urlEncode("val")` returns a future which creates a string with the characters escaped
+  so it is a valid part of a URL. The parameter can be a string *or* another task. For example,
+  `$brooklyn:urlEncode($brooklyn:config(\"mykey\"))`
```

Comment:
Let me repeat my concerns about this functionality here - see https://github.com/apache/brooklyn-server/pull/551#issuecomment-278158142.
At the very least can you document that this is meant only for query parameters. While not perfect it can do the job until improved. The various parts of the URL have different rules for escaping so this won't work universally.